src/Query/FetchProductsQuery.php

```diff
@@ -79,7 +49,7 @@ public function fetch(string $locale, array $productIdentifiers): array
 
             $values = [];
 
-            $products[] = new Product($rawProduct['identifier'], $label, $values);
+            $products[] = new Product($rawProduct['uuid'], $label, $values);
         }
 
         return $products;
```

Comment:
Can you check for `if (isset($rawProduct['message']) || isset($rawProduct['error'])) {` in this query too ?